src/purchases.ts

```diff
@@ -218,15 +245,35 @@ export default class Purchases {
       throw new Error("appUserID needs to be a string");
     }
 
+    let purchasesCompletedByToUse: PURCHASES_ARE_COMPLETED_BY_TYPE =
+      PURCHASES_ARE_COMPLETED_BY_TYPE.REVENUECAT;
+    let storeKitVersionToUse = storeKitVersion;
+
+    if (Purchases.isPurchasesAreCompletedByMyApp(purchasesAreCompletedBy)) {
+      purchasesCompletedByToUse = PURCHASES_ARE_COMPLETED_BY_TYPE.MY_APP;
+      storeKitVersionToUse = purchasesAreCompletedBy.storeKitVersion;
+
+      if (storeKitVersionToUse !== storeKitVersion) {
```

Comment:
I was just checking this out for cordova and I was thinking... Wouldn't this log if leaving `storeKitVersion` to the default value `STOREKIT_VERSION.DEFAULT` and using a non-default value when using `PurchasesAreCompletedByMyApp`? We probably should either:
- Remove the default value from `storeKitVersion`, so we know the developer is passing in a value. If it's null/undefined, no logging.
- Not log if the `storeKitVersion` passed as a parameter is `DEFAULT`

It might also be a good idea to log if someone is passing `DEFAULT` to the `PurchasesAreCompletedByMyApp` type... cc @fire-at-will